### PR TITLE
Update eventing helloworld-python dependencies

### LINKED ISSUE #4833: helloworld-python fails due legacy Python dependencies
Currently running the [helloword-python](https://github.com/knative/docs/tree/main/code-samples/eventing/helloworld/helloworld-python) example produces a container that fails with following error:

```bash
Traceback (most recent call last):
  File "helloworld.py", line 2, in <module>
    from flask import Flask, request, make_response
  File "/usr/local/lib/python3.7/site-packages/flask/__init__.py", line 19, in <module>
    from . import json
  File "/usr/local/lib/python3.7/site-packages/flask/json/__init__.py", line 15, in <module>
    from itsdangerous import json as _json
ImportError: cannot import name 'json' from 'itsdangerous' (/usr/local/lib/python3.7/site-packages/itsdangerous/__init__.py)
```

Maybe some of the dependencies are deprecated recently etc

**Describe the change you'd like to see**

Update dependencies.

It seems that updating Flask version to `2.0.1` and Dockerfile base image to `python:alpine3.9` solves the problem.

**Additional context**

### PR DESCRIPTION
**Pull request by samutamm**
Fixes #4833. 

## Proposed Changes 

- upgrade Flask to 2.0.1 in requirements.txt
- update base image to  `python:alpine3.9` in Dockerfile

### CODE REVIEW

File: code-samples/eventing/helloworld/helloworld-python/requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-Flask==1.1.1
+Flask==2.0.1
```
Comment (psschwei) [inline code-samples/eventing/helloworld/helloworld-python/requirements.txt:1]: Latest version seems to be [`2.0.3`](https://pypi.org/project/Flask/#history) ... should we go ahead and bump to the latest?

File: code-samples/eventing/helloworld/helloworld-python/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM python:alpine3.7
+FROM python:alpine3.9
```
Comment (psschwei) [inline code-samples/eventing/helloworld/helloworld-python/Dockerfile:1]: The alpine3.9 tag seems to be [out of date](https://hub.docker.com/layers/python/library/python/alpine3.9/images/sha256-442fe94d3b444303cdbb730cb56c99144d19466262249825d5b16c54ae02a265?context=explore), I think the tag that gets updates is [`3.9-alpine`](https://hub.docker.com/layers/python/library/python/3.9-alpine/images/sha256-1f078ebd3ad2405efb6dd69944046e135db45788280e1c334d10e7bf822ea842?context=explore)

**Review by psschwei — COMMENTED** [review]
Thanks for fixing this! Couple of questions on versioning below...

File: code-samples/eventing/helloworld/helloworld-python/requirements.txt (hunk 1 above)
Comment (samutamm) [reply code-samples/eventing/helloworld/helloworld-python/requirements.txt]: Updated, the latest 2.0.3 works also

File: code-samples/eventing/helloworld/helloworld-python/Dockerfile (hunk 2 above)
Comment (samutamm) [reply code-samples/eventing/helloworld/helloworld-python/Dockerfile]: Oh thanks! I changed to the active tag [3.9-alpine](https://hub.docker.com/layers/python/library/python/3.9-alpine/images/sha256-1f078ebd3ad2405efb6dd69944046e135db45788280e1c334d10e7bf822ea842?context=explore).

**Review by psschwei — COMMENTED** [review]
Thanks again for the update!

/lgtm
/approve

**Pull request merged**